src/SqlAsyncCollector.cs

```diff
@@ -543,7 +581,11 @@ public static async Task<TableInformation> RetrieveTableInformationAsync(SqlConn
                 // Match SQL Primary Key column names to POCO field/property objects. Ensure none are missing.
                 StringComparison comparison = caseSensitive ? StringComparison.Ordinal : StringComparison.OrdinalIgnoreCase;
                 IEnumerable<MemberInfo> primaryKeyFields = typeof(T).GetMembers().Where(f => primaryKeys.Any(k => string.Equals(k.Name, f.Name, comparison)));
-                IEnumerable<string> primaryKeysFromPOCO = primaryKeyFields.Select(f => f.Name);
+                if (!columnNames.Any())
```

Comment:
Add a comment about what this is doing, but also should we even be doing the logic on line 583 for JObjects? It seems like we should just be having completely separate code paths for JObject and other types.